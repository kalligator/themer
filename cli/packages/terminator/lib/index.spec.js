const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('render', () => {
  const promisedFiles = Promise.all(render(colors));

  it('should render the expected number of files', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
    expect(files.some((file) => /dark/.test(file.name))).toBe(true);
    expect(files.some((file) => /light/.test(file.name))).toBe(true);
  });

  it('should render valid terminator theme config', async () => {
    const files = await promisedFiles;
    files.forEach((file) => {
      const contents = file.contents.toString('utf8');
      [
        'background_color',
        'cursor_color',
        'foreground_color',
        'palette',
      ].forEach((prop) => {
        expect(contents).toMatch(new RegExp(prop, 'i'));
      });
    });
  });
});

describe('renderInstructions', () => {
  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
