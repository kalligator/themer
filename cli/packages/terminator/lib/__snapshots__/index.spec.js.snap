// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderInstructions should provide installation instructions 1`] = `
"
Copy the contents of \`themer-terminator-dark.txt\` or \`themer-terminator-light.txt\` to the Terminator's config file.

The config file is usually located at \`~/.config/terminator/config\`.

You can paste it as a new profile, or copy the contents into your existing profile.

Finally, restart Terminator to see the new theme.
"
`;
